src/provider/hyperkzg.rs

```diff
@@ -55,25 +62,28 @@ pub struct EvaluationEngine<E, NE> {
 impl<E, NE> EvaluationEngine<E, NE>
 where
   E: Engine,
-  NE: NovaEngine<GE = E::G1, Scalar = E::Fr>,
-  E::G1: DlogGroup,
+  NE: NovaEngine<GE = E::G1, Scalar = E::Fr, CE = KZGCommitmentEngine<E>>,
+  E::Fr: Serialize + DeserializeOwned,
+  E::G1Affine: Serialize + DeserializeOwned,
+  E::G2Affine: Serialize + DeserializeOwned,
+  E::G1: DlogGroup<ScalarExt = E::Fr, AffineExt = E::G1Affine>,
+  E::Fr: PrimeFieldBits,
+  <E::G1 as Group>::Base: TranscriptReprTrait<E::G1>,
```

Comment:
A couple of points on this smorgasboard of constraints:
- the bounds on `E::Fr` and `E::G1Affine` are implied by that on `E::G1`, and should be auto-satisfied. They should no longer be needed when [`associated_type_bounds`](https://rust-lang.github.io/rfcs/2289-associated-type-bounds.html) are possible.
- we probably want to put first the two sets of bounds that are actually defining what we need, then comment on what's going on with the rest:
```rust
  E: Engine,
  NE: NovaEngine<GE = E::G1, Scalar = E::Fr, CE = KZGCommitmentEngine<E>>,
  E::G1: DlogGroup<ScalarExt = E::Fr, AffineExt = E::G1Affine>,
  // the following bounds repeat existing, satisfied bounds on associated types of the above
  // but are required since the equality constraints we use in the above do not transitively carry bounds
  // we should be able to remove most of those constraints when rust supports associated_type_bounds
  E::Fr: Serialize + DeserializeOwned,
  E::G1Affine: Serialize + DeserializeOwned,
  E::G2Affine: Serialize + DeserializeOwned,
  E::Fr: PrimeFieldBits,
  <E::G1 as Group>::Base: TranscriptReprTrait<E::G1>,
```